lib/box-node-sdk.js

```diff
@@ -117,6 +117,36 @@ BoxSDKNode.accessLevels = BoxSDKNode.prototype.accessLevels = BoxClient.prototyp
 BoxSDKNode.collaborationRoles = BoxSDKNode.prototype.collaborationRoles = BoxClient.prototype.collaborationRoles;
 BoxSDKNode.CURRENT_USER_ID = BoxSDKNode.prototype.CURRENT_USER_ID = BoxClient.prototype.CURRENT_USER_ID;
 
+BoxSDKNode.getInstance = function(appConfig) {
+	if (typeof appConfig.boxAppSettings !== 'object') {
+		throw new Error('Configuration does not include boxAppSettings object.');
+	}
+
+	var boxAppSettings = appConfig.boxAppSettings;
+	if (typeof boxAppSettings.webhooks === 'object') {
+
+		var primaryKey = boxAppSettings.webhooks.primaryKey;
+		if (typeof primaryKey === 'string') {
+			Webhooks.primaryKey = primaryKey;
+		}
+
+		var secondaryKey = boxAppSettings.webhooks.secondaryKey;
+		if (typeof secondaryKey === 'string') {
+			Webhooks.secondaryKey = secondaryKey;
+		}
+	}
+
+	//Assign publicKeyID to keyID
+	if (typeof boxAppSettings.appAuth === 'object') {
+		var keyID = boxAppSettings.appAuth.publicKeyID;
+		if (typeof keyID === 'string') {
+			boxAppSettings.appAuth.keyID = keyID;
+		}
+	}
+
+	return new BoxSDKNode(boxAppSettings);
```

Comment:
Rather than passing this object through directly, it might be better to construct our own object by pulling things off of `boxAppSettings`, just to make sure no random properties make it through that we don't expect.